appcypher/monocore
# refactor(config)!: simplify service model and improve volume handling

# Description

This PR refactors the configuration system to simplify the service model and improve volume handling. The main change consolidates the three service types (Default, HttpHandler, Precursor) into a single unified Service struct, while maintaining all existing functionality. This simplification makes the configuration model more maintainable and easier to understand.

Key changes:
- Unified service model with direct volume and environment variable support
- Improved volume path normalization and validation
- New helper methods for resolving volumes and environment variables
- Updated test suite to cover new configuration model
- Dependency updates including new camino and pretty-error-debug crates

## Link to issue

- Part of larger effort to improve DX 
    - #51
    - #55
    - #56

## Type of change

- [x] Comments have been added/updated

## Test plan (required)

The changes are covered by extensive unit tests that verify:
- Service configuration parsing from both TOML and JSON
- Environment variable resolution with proper precedence
- Volume path normalization and validation
- Group volume mounting and validation
- Service dependency ordering
- Configuration merging with conflict detection

All existing tests have been updated to use the new unified service model. New tests have been added to cover:
- Volume path normalization and conflict detection
- Environment variable resolution and overriding
- Group volume mounting with subdirectories
- Configuration validation for the new service model

To test manually:
1. Create a configuration using the new unified service model
2. Verify volume mounting works with both direct and group volumes
3. Verify environment variables are properly resolved from both service and group levels
4. Test configuration merging with the new validation rules

## Screenshots/Screencaps

N/A
